activerecord/lib/active_record/associations/builder/collection_association.rb

```diff
@@ -66,17 +66,5 @@ def #{name.to_s.singularize}_ids=(ids)
         end
       CODE
     end
-
-    def self.wrap_scope(scope, mod)
```

Comment:
I changed `def has_many(name, scope = nil, options = {}, &extension)` to `def has_many(name, scope = nil, **options, &extension)` to don't mutate passed `options`. And then do `options[:extend] = Array(options[:extend]) + [extension] if extension`.
Is it not enough?

https://github.com/rails/rails/pull/28830/files#diff-a215bfc911ab0913d92695a0f20310efR1395
https://github.com/rails/rails/pull/28830/files#diff-5870816b49b90e43340607bb11ed2514R41